src/router/index.js

```diff
@@ -0,0 +1,24 @@
+import { createRouter, createWebHistory } from "vue-router";
+import HomePage from "../pages/Homepage.vue";
+
+const routes = [
+    {
+        path: "/",
+        name:"HomePage",
+        component: HomePage,
+      //   props: (route) => ({
```

Comment:
No params yet?

Reply:
I'm actually directly doing this => this.$router.push(this.$route.query.redirecturl); (or this.$route.query.purpose)
@deepansh96 should let me know if its alright, this way.